lib/core/src/receive_swap.rs

```diff
@@ -270,6 +301,78 @@ impl ReceiveSwapStateHandler {
             (_, Failed) => Ok(()),
         }
     }
+
+    async fn verify_lockup_tx(
+        &self,
+        receive_swap: &ReceiveSwap,
+        swap_update_tx: &SwapUpdateTxDetails,
+    ) -> Result<()> {
+        // looking for lockup script history to verify lockup was broadcasted
+        let script_history = self.lockup_script_history(receive_swap).await?;
+        let lockup_tx_history = script_history
+            .iter()
+            .find(|h| h.txid.to_hex().eq(&swap_update_tx.id));
+
+        if lockup_tx_history.is_none() {
```

Comment:
Might be better to replace this with a
```rust
match lockup_tx_history {
    None => {

    }
    Some(history) => {

    }
}
```

to make various cases simpler to handle, and to get rid of the `unwrap` below.

Reply:
fixed